sky/clouds/cloud.py

```diff
@@ -489,13 +490,41 @@ def get_current_user_identity(cls) -> Optional[List[str]]:
         return None
 
     @classmethod
-    def get_current_user_identity_str(cls) -> Optional[str]:
-        """Returns a user friendly representation of the current identity."""
-        user_identity = cls.get_current_user_identity()
+    def get_active_user_identity_str(cls) -> Optional[str]:
+        """Returns a user friendly representation of the active identity."""
+        user_identity = cls.get_active_user_identity()
         if user_identity is None:
             return None
         return ', '.join(user_identity)
 
+    @classmethod
+    def get_user_identities(cls) -> Optional[List[List[str]]]:
+        """Returns all available user identities of this cloud.
+
+        See get_active_user_identity for definition of user identity.
+
+        This method returns a list of user identities, with the current active
+        identity being the first element. Most clouds have only one identity
+        available, so the returned list will only have one element: the current
+        active identity.
+
+        However, some clouds (e.g., Kubernetes) can have multiple current
+        identities (e.g., multiple contexts configured in kubeconfig) that
+        can be dynamically switched, so the list can have multiple elements.
+
+        Example return values:
+            - AWS: [[UserId, AccountId]]
+            - GCP: [[email address + project ID]]
+            - Azure: [[email address + subscription ID]]
+            - Kubernetes: [[current active context], [context 2], ...]
+
+        Returns:
+            None if the cloud does not have a concept of user identity;
+            otherwise all the user identities.
+        """
+        active_identity = cls.get_active_user_identity()
+        return [active_identity] if active_identity is not None else None
```

Comment:
I found calling `get_active_user_identity` within `get_user_identities` a bit weird to me. Should we instead make `get_user_identities` always return a list of identities with the active one being the first one, and having `get_active_user_identity` to index the returned list of identities? 

With that, we can also add `TODO` in AWS and GCP's implementation: `Return a list of identities in the profile, when we support automatically switching context.`

Reply:
I see what you mean, fixed now!